File: scratch/find_chinese.js
const fs = require('fs');
const path = require('path');

const TOOLBOX_DIR = path.join(__dirname, '../src/app/toolbox');

function scanDir(dir) {
  let results = [];
  const files = fs.readdirSync(dir);
  for (const file of files) {
    const fullPath = path.join(dir, file);
    const stat = fs.statSync(fullPath);
    if (stat.isDirectory()) {
      results = results.concat(scanDir(fullPath));
    } else if (file.endsWith('.tsx') || file.endsWith('.ts')) {
      results.push(fullPath);
    }
  }
  return results;
}

const files = scanDir(TOOLBOX_DIR);
const chineseRegex = /[\u4e00-\u9fa5]/;

let reportText = '--- SCANNING FOR CHINESE CHARACTERS ---\n';
let foundCount = 0;

for (const file of files) {
  const content = fs.readFileSync(file, 'utf8');
  if (chineseRegex.test(content)) {
    const relative = path.relative(path.join(__dirname, '..'), file);
    reportText += `Found Chinese in: ${relative}\n`;
    foundCount++;
    
    // Print lines containing Chinese
    const lines = content.split('\n');
    lines.forEach((line, index) => {
      if (chineseRegex.test(line)) {
        reportText += `  L${index + 1}: ${line.trim()}\n`;
      }
    });
    reportText += '\n';
  }
}

reportText += `Scan complete. Found Chinese characters in ${foundCount} files.\n`;
fs.writeFileSync(path.join(__dirname, 'chinese_report.txt'), reportText);
console.log('Report written to scratch/chinese_report.txt');
console.log(`Found Chinese characters in ${foundCount} files.`);

File: scratch/find_chinese.py
import os
import re

# Regex to match Chinese characters
chinese_regex = re.compile(r'[\u4e00-\u9fa5]')

root_dir = "F:/cad tools io/cadtools-cc/src"

results = []

for dirpath, _, filenames in os.walk(root_dir):
    for filename in filenames:
        if filename.endswith(('.ts', '.tsx')):
            filepath = os.path.join(dirpath, filename)
            try:
                with open(filepath, 'r', encoding='utf-8') as f:
                    for line_num, line in enumerate(f, 1):
                        if chinese_regex.search(line):
                            results.append({
                                'file': filepath,
                                'line_num': line_num,
                                'content': line.strip()
                            })
            except Exception as e:
                print(f"Error reading {filepath}: {e}")

print(f"Found {len(results)} lines containing Chinese characters:")
for r in results:
    # Print in relative format for readability
    rel_path = os.path.relpath(r['file'], "F:/cad tools io/cadtools-cc")
    print(f"{rel_path}:{r['line_num']}: {r['content']}")
